dataset/__init__.py

```diff
@@ -173,6 +181,12 @@ def get_normalized_dataset(dataset, ps, seeds, d=0):
         if dataset == 'sphere':
             r = x.norm(dim=1)
             y = (r**2 > d - 2 / 3)
+        if dataset == 'cylinder':
+            dsph = d // 2 if params[0] is None else params[0]
+            stretching = 1 if params[1] is None else params[1]
```

Comment:
Je te conseil pas d'avoir des paramètres par défaut ici
```suggestion
            dsph = params[0]
            stretching = params[1]
            assert dsph is not None and stretching is not None
```